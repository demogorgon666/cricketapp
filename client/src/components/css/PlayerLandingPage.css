@import url("https://fonts.googleapis.com/css?family=Montserrat:400,500,600&display=swap");

.matchFormatOptons {
  margin-left: 20px;
  margin-top: 30px;
  border: 1px solid #16a085;
  border-radius: 8px;
  width: 237px;
}

.selectNav a {
  float: left;
  color: #16a085;
  text-align: center;
  padding: 12px 14px;
  text-decoration: none;
  font-size: 16px;
  border-right: 1px solid #16a085;
  font-family: "Monsterrat";
  /* border-top-right-radius: 8px; */
  /* border-radius: 8px; */
  width: 50px;
}

.selectNav a:hover {
  background-color: #16a085;
  color: white;
}

.selectNav a.active {
  background-color: #16a085;
  color: white;
}

#conatiner {
  display: flex;
  flex-direction: column;
  margin: 20px;
}
@media only screen and (max-width: 1500px) {
  .allPlayersDiv {
    display: flex;
    flex-direction: row;
    margin-left: 20px;
    margin-top: 2em;
  }
}
@media only screen and (max-width: 1500px) {
  .topbatsmanDiv {
    display: flex;
    flex-direction: column;
    border: 1px solid #707070;
    border-radius: 8px;
    max-height: 800px;
    width: 600px;
    margin: 8x;
    font-family: "Montserrat";
  }
}

.batsmenDetailDiv {
  display: flex;
  flex-direction: row;
  margin: 10px;
  justify-content: space-between;
  color: #383838;
  align-items: center;
  /* justify-content: s */
}

hr {
  margin: auto;
  width: 80%;
}

.playerImage {
  height: 100px;
  width: 100px;
  border-radius: 50%;
  border: 1px solid #707070;
}
@media only screen and (max-width: 1500px) {
  .topbowlerDiv {
    display: flex;
    flex-direction: column;
    border: 1px solid #707070;
    border-radius: 8px;
    max-height: 800px;
    width: 600px;
    margin-left: 2em;
    font-family: "Montserrat";
  }
}
