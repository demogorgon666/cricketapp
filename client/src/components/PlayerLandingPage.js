import React, { Component } from "react";

import virat from "../components/images/virat.jpg";
import bumrah from "../components/images/Bumrah.jpg";

import "./css/PlayerLandingPage.css";
import NavBar from "../components/common/Navbar";

export default class PlayerLandingPage extends Component {
  render() {
    return (
      <div style={{ padding: 12 + "px" }}>
        <NavBar></NavBar>
        <h1
          style={{
            textAlign: "left",
            marginLeft: 35 + "px",
            marginTop: 2 + "em",
            fontSize: 42 + "px",
            fontWeight: 500,
            fontFamily: "'Montserrat'"
          }}
        >
          Players
        </h1>
        <div id="conatiner">
          <div className="matchFormatOptons">
            <div className="selectNav">
              <a
                href="#Test"
                className="active"
                style={{
                  borderLeft: "none",
                  borderTopLeftRadius: 8 + "px",
                  borderBottomLeftRadius: 8 + "px"
                }}
              >
                Test
              </a>

              <a href="#Odi">ODI</a>

              <a
                href="#T20"
                style={{
                  borderRight: "none"
                }}
              >
                T20
              </a>
            </div>
          </div>
          <div className="allPlayersDiv">
            <div className="topbatsmanDiv">
              <div className="insideBatsmanDiv">
                <h2 style={{ textAlign: "center" }}>Top Batsmen</h2>
                <div className="batsmenDetailDiv">
                  <img src={virat} alt="virat kohli" className="playerImage" />
                  <div
                    style={{
                      display: "flex",
                      flexDirection: "column",
                      marginTop: 10 + "px",
                      textAlign: "left"
                    }}
                  >
                    <div
                      style={{
                        display: "flex",
                        flexDirection: "column"
                      }}
                    >
                      <span>
                        <b>Virat Kohli</b>
                      </span>
                      <span>Right-Hand Batsman</span>
                    </div>
                  </div>
                  <div style={{ display: "flex", flexDirection: "column" }}>
                    <b
                      style={{
                        justifySelf: "flex-end",
                        fontSize: 25 + "px",
                        textAlign: "right"
                      }}
                    >
                      672
                    </b>
                    <span style={{ textAlign: "right" }}>Innings: 12</span>
                    <span>Average: 52.12</span>
                  </div>
                </div>
                <hr></hr>
                <div className="batsmenDetailDiv">
                  <img src={virat} alt="virat kohli" className="playerImage" />
                  <div
                    style={{
                      display: "flex",
                      flexDirection: "column",
                      marginTop: 10 + "px",

                      textAlign: "left"
                    }}
                  >
                    <span>
                      <b>Virat Kohli</b>
                    </span>
                    <span>Right-Hand Batsman</span>
                  </div>
                  <div style={{ display: "flex", flexDirection: "column" }}>
                    <b
                      style={{
                        justifySelf: "flex-end",
                        fontSize: 25 + "px",
                        textAlign: "right"
                      }}
                    >
                      672
                    </b>
                    <span style={{ textAlign: "right" }}>Innings: 12</span>
                    <span style={{ textAlign: "right" }}>Average: 52.12</span>
                  </div>
                </div>
                <hr></hr>
                <div className="batsmenDetailDiv">
                  <img src={virat} alt="virat kohli" className="playerImage" />
                  <div
                    style={{
                      display: "flex",
                      flexDirection: "column",
                      marginTop: 10 + "px",

                      textAlign: "left"
                    }}
                  >
                    <span>
                      <b>Virat Kohli</b>
                    </span>
                    <span>Right-Hand Batsman</span>
                  </div>
                  <div style={{ display: "flex", flexDirection: "column" }}>
                    <b
                      style={{
                        justifySelf: "flex-end",
                        fontSize: 25 + "px",
                        textAlign: "right"
                      }}
                    >
                      672
                    </b>
                    <span style={{ textAlign: "right" }}>Innings:12</span>
                    <span style={{ textAlign: "right" }}>Average:52.12</span>
                  </div>
                </div>
              </div>
            </div>
            <div className="topbowlerDiv">
              <div className="insideBowlerDiv">
                <h2 style={{ textAlign: "center" }}>Top Bowlers</h2>
                <div className="batsmenDetailDiv">
                  <img src={bumrah} alt="virat kohli" className="playerImage" />
                  <div
                    style={{
                      display: "flex",
                      flexDirection: "column",
                      marginTop: 10 + "px",
                      textAlign: "left"
                    }}
                  >
                    <span>
                      <b>Jasprit Bumrah</b>
                    </span>
                    <span>Right-handed fast Bowler</span>
                  </div>
                  <div style={{ display: "flex", flexDirection: "column" }}>
                    <b
                      style={{
                        justifySelf: "flex-end",
                        fontSize: 25 + "px",
                        textAlign: "right"
                      }}
                    >
                      93
                    </b>
                    <span style={{ textAlign: "right" }}>Innings: 12</span>
                    <span style={{ textAlign: "right" }}>E/R: 6.2</span>
                  </div>
                </div>
                <hr></hr>
                <div className="batsmenDetailDiv">
                  <img src={bumrah} alt="virat kohli" className="playerImage" />
                  <div
                    style={{
                      display: "flex",
                      flexDirection: "column",
                      marginTop: 10 + "px",
                      textAlign: "left"
                    }}
                  >
                    <span>
                      <b>Jasprit Bumrah</b>
                    </span>
                    <span>Right-handed fast Bowler</span>
                  </div>
                  <div style={{ display: "flex", flexDirection: "column" }}>
                    <b
                      style={{
                        justifySelf: "flex-end",
                        fontSize: 25 + "px",
                        textAlign: "right"
                      }}
                    >
                      93
                    </b>
                    <span style={{ textAlign: "right" }}>Innings: 12</span>
                    <span style={{ textAlign: "right" }}>E/R: 6.2</span>
                  </div>
                </div>
                <hr></hr>
                <div className="batsmenDetailDiv">
                  <img src={bumrah} alt="virat kohli" className="playerImage" />
                  <div
                    style={{
                      display: "flex",
                      flexDirection: "column",
                      marginTop: 10 + "px",

                      textAlign: "left"
                    }}
                  >
                    <span>
                      <b>Jasprit Bumrah</b>
                    </span>
                    <span>Right-handed fast Bowler</span>
                  </div>
                  <div style={{ display: "flex", flexDirection: "column" }}>
                    <b
                      style={{
                        justifySelf: "flex-end",
                        fontSize: 25 + "px",
                        textAlign: "right"
                      }}
                    >
                      93
                    </b>
                    <span style={{ textAlign: "right" }}>Innings: 12</span>
                    <span style={{ textAlign: "right" }}>E/R: 6.2</span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}
