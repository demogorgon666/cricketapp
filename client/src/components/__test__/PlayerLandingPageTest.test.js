import React from "react";
import { shallow, mount } from "enzyme";
import { PlayerLandingPage } from "../PlayerLandingPage";

const getBowlers = jest.fn();
var bowlers = [];

const getBatsmen = jest.fn();
var batsmen = [];

const wrapper = shallow(
  <PlayerLandingPage
    getBowlers={getBowlers}
    bowlers={bowlers}
    getBatsmen={getBatsmen}
    batsmen={batsmen}
  />
);

describe("Testing Player Landing Page", () => {
  it("should render the component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should have 9 div", () => {
    expect(wrapper.find("div").length).toBe(9);
  });

  it("should have the name of header as Players", () => {
    expect(wrapper.find("h1").text()).toBe("Players");
  });

  it("should have links for Test,Odi and T20", () => {
    expect(
      wrapper
        .find("a")
        .at(0)
        .text()
    ).toBe("Test");

    expect(
      wrapper
        .find("a")
        .at(1)
        .text()
    ).toBe("ODI");

    expect(
      wrapper
        .find("a")
        .at(2)
        .text()
    ).toBe("T20");
  });

  it("should have a header for top bowlers", () => {
    expect(
      wrapper
        .find("h2")
        .at(1)
        .text()
    ).toBe("Top Bowlers");
  });

  it("should have a header for top batsman", () => {
    expect(
      wrapper
        .find("h2")
        .at(0)
        .text()
    ).toBe("Top Batsmen");
  });

  it("should display 3 select menus", () => {
    expect(wrapper.find("a").length).toBe(3);
  });
});
