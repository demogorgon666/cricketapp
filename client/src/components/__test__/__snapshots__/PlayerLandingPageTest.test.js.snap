// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Player Landing Page should render the component 1`] = `
<div
  style={
    Object {
      "padding": "12px",
    }
  }
>
  <Navbar />
  <h1
    style={
      Object {
        "fontFamily": "'Montserrat'",
        "fontSize": "42px",
        "fontWeight": 500,
        "marginLeft": "35px",
        "marginTop": "2em",
        "textAlign": "left",
      }
    }
  >
    Players
  </h1>
  <div
    id="conatiner"
  >
    <div
      className="matchFormatOptons"
    >
      <div
        className="selectNav"
      >
        <a
          className="active"
          href="#Test"
          style={
            Object {
              "borderBottomLeftRadius": "8px",
              "borderLeft": "none",
              "borderTopLeftRadius": "8px",
            }
          }
        >
          Test
        </a>
        <a
          href="#Odi"
        >
          ODI
        </a>
        <a
          href="#T20"
          style={
            Object {
              "borderRight": "none",
            }
          }
        >
          T20
        </a>
      </div>
    </div>
    <div
      className="allPlayersDiv"
    >
      <div
        className="topbatsmanDiv"
      >
        <div
          className="insideBatsmanDiv"
        >
          <h2
            style={
              Object {
                "textAlign": "center",
              }
            }
          >
            Top Batsmen
          </h2>
          <div
            className="batsmenDetailDiv"
          >
            <img
              alt="virat kohli"
              className="playerImage"
              src="virat.jpg"
            />
            <div
              style={
                Object {
                  "display": "flex",
                  "flexDirection": "column",
                  "marginTop": "10px",
                  "textAlign": "left",
                }
              }
            >
              <div
                style={
                  Object {
                    "display": "flex",
                    "flexDirection": "column",
                  }
                }
              >
                <span>
                  <b>
                    Virat Kohli
                  </b>
                </span>
                <span>
                  Right-Hand Batsman
                </span>
              </div>
            </div>
            <div
              style={
                Object {
                  "display": "flex",
                  "flexDirection": "column",
                }
              }
            >
              <b
                style={
                  Object {
                    "fontSize": "25px",
                    "justifySelf": "flex-end",
                    "textAlign": "right",
                  }
                }
              >
                672
              </b>
              <span
                style={
                  Object {
                    "textAlign": "right",
                  }
                }
              >
                Innings: 12
              </span>
              <span>
                Average: 52.12
              </span>
            </div>
          </div>
          <hr />
          <div
            className="batsmenDetailDiv"
          >
            <img
              alt="virat kohli"
              className="playerImage"
              src="virat.jpg"
            />
            <div
              style={
                Object {
                  "display": "flex",
                  "flexDirection": "column",
                  "marginTop": "10px",
                  "textAlign": "left",
                }
              }
            >
              <span>
                <b>
                  Virat Kohli
                </b>
              </span>
              <span>
                Right-Hand Batsman
              </span>
            </div>
            <div
              style={
                Object {
                  "display": "flex",
                  "flexDirection": "column",
                }
              }
            >
              <b
                style={
                  Object {
                    "fontSize": "25px",
                    "justifySelf": "flex-end",
                    "textAlign": "right",
                  }
                }
              >
                672
              </b>
              <span
                style={
                  Object {
                    "textAlign": "right",
                  }
                }
              >
                Innings: 12
              </span>
              <span
                style={
                  Object {
                    "textAlign": "right",
                  }
                }
              >
                Average: 52.12
              </span>
            </div>
          </div>
          <hr />
          <div
            className="batsmenDetailDiv"
          >
            <img
              alt="virat kohli"
              className="playerImage"
              src="virat.jpg"
            />
            <div
              style={
                Object {
                  "display": "flex",
                  "flexDirection": "column",
                  "marginTop": "10px",
                  "textAlign": "left",
                }
              }
            >
              <span>
                <b>
                  Virat Kohli
                </b>
              </span>
              <span>
                Right-Hand Batsman
              </span>
            </div>
            <div
              style={
                Object {
                  "display": "flex",
                  "flexDirection": "column",
                }
              }
            >
              <b
                style={
                  Object {
                    "fontSize": "25px",
                    "justifySelf": "flex-end",
                    "textAlign": "right",
                  }
                }
              >
                672
              </b>
              <span
                style={
                  Object {
                    "textAlign": "right",
                  }
                }
              >
                Innings:12
              </span>
              <span
                style={
                  Object {
                    "textAlign": "right",
                  }
                }
              >
                Average:52.12
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        className="topbowlerDiv"
      >
        <div
          className="insideBowlerDiv"
        >
          <h2
            style={
              Object {
                "textAlign": "center",
              }
            }
          >
            Top Bowlers
          </h2>
          <div
            className="batsmenDetailDiv"
          >
            <img
              alt="virat kohli"
              className="playerImage"
              src="Bumrah.jpg"
            />
            <div
              style={
                Object {
                  "display": "flex",
                  "flexDirection": "column",
                  "marginTop": "10px",
                  "textAlign": "left",
                }
              }
            >
              <span>
                <b>
                  Jasprit Bumrah
                </b>
              </span>
              <span>
                Right-handed fast Bowler
              </span>
            </div>
            <div
              style={
                Object {
                  "display": "flex",
                  "flexDirection": "column",
                }
              }
            >
              <b
                style={
                  Object {
                    "fontSize": "25px",
                    "justifySelf": "flex-end",
                    "textAlign": "right",
                  }
                }
              >
                93
              </b>
              <span
                style={
                  Object {
                    "textAlign": "right",
                  }
                }
              >
                Innings: 12
              </span>
              <span
                style={
                  Object {
                    "textAlign": "right",
                  }
                }
              >
                E/R: 6.2
              </span>
            </div>
          </div>
          <hr />
          <div
            className="batsmenDetailDiv"
          >
            <img
              alt="virat kohli"
              className="playerImage"
              src="Bumrah.jpg"
            />
            <div
              style={
                Object {
                  "display": "flex",
                  "flexDirection": "column",
                  "marginTop": "10px",
                  "textAlign": "left",
                }
              }
            >
              <span>
                <b>
                  Jasprit Bumrah
                </b>
              </span>
              <span>
                Right-handed fast Bowler
              </span>
            </div>
            <div
              style={
                Object {
                  "display": "flex",
                  "flexDirection": "column",
                }
              }
            >
              <b
                style={
                  Object {
                    "fontSize": "25px",
                    "justifySelf": "flex-end",
                    "textAlign": "right",
                  }
                }
              >
                93
              </b>
              <span
                style={
                  Object {
                    "textAlign": "right",
                  }
                }
              >
                Innings: 12
              </span>
              <span
                style={
                  Object {
                    "textAlign": "right",
                  }
                }
              >
                E/R: 6.2
              </span>
            </div>
          </div>
          <hr />
          <div
            className="batsmenDetailDiv"
          >
            <img
              alt="virat kohli"
              className="playerImage"
              src="Bumrah.jpg"
            />
            <div
              style={
                Object {
                  "display": "flex",
                  "flexDirection": "column",
                  "marginTop": "10px",
                  "textAlign": "left",
                }
              }
            >
              <span>
                <b>
                  Jasprit Bumrah
                </b>
              </span>
              <span>
                Right-handed fast Bowler
              </span>
            </div>
            <div
              style={
                Object {
                  "display": "flex",
                  "flexDirection": "column",
                }
              }
            >
              <b
                style={
                  Object {
                    "fontSize": "25px",
                    "justifySelf": "flex-end",
                    "textAlign": "right",
                  }
                }
              >
                93
              </b>
              <span
                style={
                  Object {
                    "textAlign": "right",
                  }
                }
              >
                Innings: 12
              </span>
              <span
                style={
                  Object {
                    "textAlign": "right",
                  }
                }
              >
                E/R: 6.2
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
